File: swift/quickSort.swift
// クイックソート
func quickSort(a: [Int]) -> [Int] {
    var arr = a
    func q(head: Int, tail: Int) -> [Int] {
        var i = head
        var j = tail
        var pivot = arr[Int(floor(Double(head + tail) / 2))]
        while true {
            while arr[i] < pivot {
                i += 1
            }
            while pivot < arr[j] {
                j -= 1
            }
            if i >= j {
                break;
            }
            var tmp = arr[i]
            arr[i] = arr[j]
            arr[j] = tmp
            i += 1
            j -= 1
        }
        if head < i - 1 {
            q(head: head, tail: i-1)
        }
        if j + 1 < tail {
            q(head: j+1, tail: tail)
        }
        return arr
    }
    return q(head: 0, tail: arr.count-1)
}
print(quickSort(a: [867, 104, 413, 541, 304, 354, 165, 150]))

File: php/quickSort.php
<?php
// クイックソート
function quickSort($arr) {
  function q($arr, $head, $tail) {
    $i = $head;
    $j = $tail;
    $pivot = $arr[floor(($head + $tail) / 2)];
    
    for ( ; ; ) {
      while ($arr[$i] < $pivot) {
        $i++;
      }
      while ($pivot < $arr[$j]) {
        $j--;
      }
      if ($i >= $j) {
        break;
      }
      $tmp = $arr[$i];
      $arr[$i] = $arr[$j];
      $arr[$j] = $tmp;  
      $i++;
      $j--;
    }
    if ($head < $i - 1) {
      q($arr, $head, $i-1);
    }
    if ($j + 1 < $tail) {
      q($arr, $j+1, $tail);
    }
    return $arr;
  }
  return q($arr, 0, count($arr)-1);
}
$arr = array(651, 148 , 124, 638, 567, 435, 185, 413, 841, 35);
print_r(quickSort($arr));
